### PR TITLE
Issue #15029: Update EmptyBlockCheck to Violate Empty Case and Default Blocks wit…

### LINKED ISSUE #15029: Update EmptyBlockCheck to Violate Empty Case Blocks with Arrow Syntax
child of https://github.com/checkstyle/checkstyle/issues/14961

I have read check documentation: https://checkstyle.org/checks/blocks/emptyblock.html#EmptyBlock
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```
PS D:\CS\test> javac  src/Test.java       
 PS D:\CS\test> cat  src/Test.java         
record ColoredPoint(int p, int x, String c) { }
record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }

public class Test {

    void test(Object obj) {
        int y = 0;
        switch (obj) {
            case ColoredPoint(int x, _, _) when(x >= 2) -> { }  // expected violation
            case ColoredPoint(int x, _, _) -> { }  // expected violation
            default -> { }  // expected violation
        }
        switch (obj) {
            case ColoredPoint(int x, _, _) when(x >= 2) : { } break;  // violation
            case ColoredPoint(int x, _, _) : { } // violation
            default : { }  // violation
        }
    }
}   
PS D:\CS\test> cat config.xml                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="EmptyBlockCheck">
            <property name="tokens" value="LITERAL_CASE, LITERAL_DEFAULT "/>
        </module>
    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:14:59: Must have at least one statement. [EmptyBlock]
[ERROR] D:\CS\test\src\Test.java:15:46: Must have at least one statement. [EmptyBlock]
[ERROR] D:\CS\test\src\Test.java:16:23: Must have at least one statement. [EmptyBlock]
Audit done.
Checkstyle ends with 3 errors.
PS D:\CS\test> 

```

**Describe what you want in detail**

nothing in the doc says that switch rule blocks (arrow syntax) are not validated also It can be empty so I am expecting a violation on empty blocks in the switch rule

### PR DESCRIPTION
**Pull request by mahfouz72**
closes #15029 

Diff Regression config: https://gist.githubusercontent.com/mahfouz72/afd57aa6e51ca161b7927acd959e5211/raw/8ee6d2081667934094dc86562f74ccc96f54bfb3/check.xml

### CODE REVIEW

**Comment by romani** [comment]
Please consider Checker violation.

**Comment by mahfouz72** [comment]
We usually suppress such violations. Isn’t this acceptable?

**Comment by mahfouz72** [comment]
Github, generate report

**Comment by romani** [comment]
>We usually suppress such violations. Isn’t this acceptable?

Yes and no :). We have it to not grow list of problems.

https://github.com/checkstyle/checkstyle/actions/runs/9942558796/job/27464302193?pr=15273#step:6:1718

>
```<fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java</fileName>
    <specifier>introduce.eliminate</specifier>
    <message>It is bad style to create an Optional just to chain methods to get a non-optional value.</message>
    <lineContent>.orElse(null);</lineContent>
```

@nrmancuso , will make decision to fix or extend suppression.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java [hunk 1]
```diff
@@ -113,9 +113,9 @@ void howMany7(Nums k) {
     void howMany8(Nums k) {
         switch (k) {
             case ONE -> System.out.println("case two");
-            case TWO, THREE -> {} // ok, empty blocks are allowed for cases
+            case TWO, THREE -> {} // violation, 'Must have at least one statement'
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java:116]: Text was added in https://github.com/checkstyle/checkstyle/issues/8662 . `CASE` was a token of the check then, so I am slightly confused why we originally said this was ok to be allowed.

@nrmancuso When you review.

**Comment by rnveach** [comment]
CI is failing.

It is not clear from from inputs which token this flagging under now. Is this from case / default as talked about at https://github.com/checkstyle/checkstyle/issues/15029#issuecomment-2177544141?

Regression was only run on default. It seems to me we need regression on case as well.

Since we ran regression on default, we have a spot at https://checkstyle.org/checks/blocks/emptyblock.html#Example3-config for example we can contribute to for this new update.

===

https://checkstyle.org/checks/blocks/emptyblock.html#Description
> This check does not validate sequential blocks. 

What is a sequential block?

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java:116]: > I am slightly confused why we originally said this was ok to be allowed.

me too,  and in the same file ` case : { } ` is a violation

It was added based on this review https://github.com/checkstyle/checkstyle/pull/8697#discussion_r471072392

> This behavior is (poorly) [documented](https://checkstyle.sourceforge.io/config_blocks.html#EmptyBlock), let's leave it as it is.

Comment (rnveach) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java:116]: We don't have to have an exhaustive list in these cases, do we? I could see those not needing anything in the blocks but I don't think adding documentation would be a bad idea.

**Comment by mahfouz72** [comment]
github, generate report

**Comment by mahfouz72** [comment]
> It is not clear from from inputs which token this flagging under now. Is this from case / default as talked about at https://github.com/checkstyle/checkstyle/issues/15029#issuecomment-2177544141?
 
 This PR should fix `case` and `default` with arrow syntax any other token shouldn't be affected.
 
 > What is a sequential block?
 
 I have no idea. The doc for this check is not good should we update it? in another PR? 
 
 > Also, no violations for fallthrough

  I don't get this also do you understand what is trying to say?
  
 also, we have some incorrect renderring of tags: `Verification empty block is done for single nearest {@code case} or {@code default}.
 `

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java:116]: > We don't have to have an exhaustive list in these cases

No, this is true in switch expressions only.

> I don't think adding documentation would be a bad idea.

Updating example 3 with some cases and default with arrow syntax is good

**Comment by mahfouz72** [comment]
github, generate report

**Comment by mahfouz72** [comment]
github, generate report

**Comment by mahfouz72** [comment]
Reports:

1. `LITERAL_DEFAULT` : https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9179e85_2024173139/reports/diff/index.html

- violation added for empty default blocks with arrow syntax (expected)

2. `LITERAL_CASE` : https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9179e85_2024165045/reports/diff/index.html

- violation added for empty case blocks with arrow syntax (expected)

3. All other tokens : https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/9179e85_2024165045/reports/diff/index.html

- no difference (expected)

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java:116]: Updated example 3 with more cases and default example with `->` and `:`

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    switch (a) {
+      case 1 -> { } // violation, 'Must have at least one statement'
+      default -> { } // violation, 'Must have at least one statement'
+    }
+
+    switch (a) {
+      case 1 : { }  // violation, 'Must have at least one statement'
+      default : { }  // violation, 'Must have at least one statement'
+    }
```
Comment (nrmancuso) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java:32]: Let me play devils advocate here.

Suppose that you are a developer that wrote the second case here:
```java
    switch (a) {
      case 1 : { }  // violation, 'Must have at least one statement'
      default : { }  // violation, 'Must have at least one statement'
    }
```

You can "fix" these violations like this:
```java
    switch (a) {
      case 1 :
      default :
    }
```
Now suppose you wrote this:
```java
    switch (a) {
      case 1 -> { } // violation, 'Must have at least one statement'
      default -> { } // violation, 'Must have at least one statement'
    }
```

How would you fix this code?

Comment (mahfouz72) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java:32]: I would completely remove the case label. 
Let me play in reverse and consider this similar case.

```Java
        int x = 0;
        if (x == 1) {   // violation
          
        } else {  // violation

        }
```
How would you fix this code? you can't remove the `{}` but you will remove the if statement.

Comment (mahfouz72) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java:32]: @nrmancuso ping

Comment (nrmancuso) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java:32]: After playing around with this locally, I don’t think the compiler will let us have a case where we would be in the position to place a violation on code that would fail compilation.

I was concerned that we would place a violation on code, user would remove it, then the code would not compile. I do not think this is possible.

I am good here.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 3]
```diff
@@ -271,21 +272,25 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
...
+            leftCurly = parent.findFirstToken(TokenTypes.SLIST);
         }
-        else if (slistAST == null) {
-            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        else if (parentType == TokenTypes.CASE_GROUP) {
+            final DetailAST statementList = ast.getNextSibling();
+            leftCurly = Optional.ofNullable(statementList)
+                         .map(DetailAST::getFirstChild)
+                         .filter(node -> node.getType() == TokenTypes.SLIST)
+                         .orElse(null);
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:287]: I think this is not a great usage of Optional (defaulting to a null value); it would make more sense for this method to return an Optional, if anything. How can we improve this?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: config/checker-framework-suppressions/checker-nullness-optional-interning-suppressions.xml [hunk 4]
```diff
@@ -1679,6 +1679,31 @@
...
+  </checkerFrameworkError>
+
+  <checkerFrameworkError unstable="false">
+    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java</fileName>
+    <specifier>introduce.eliminate</specifier>
+    <message>It is bad style to create an Optional just to chain methods to get a non-optional value.</message>
+    <lineContent>.orElseGet(() -&gt; ast.findFirstToken(TokenTypes.LCURLY));</lineContent>
+  </checkerFrameworkError>
+
+  <checkerFrameworkError unstable="false">
```
[the remark this answers is not in the document]
Comment (mahfouz72) [reply config/checker-framework-suppressions/checker-nullness-optional-interning-suppressions.xml]: done

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.blocks.emptyblock;
+
+// xdoc section -- start
+public class Example3 {
+
+  private void test(int a) {
```
[the remark this answers is not in the document]
Comment (mahfouz72) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/Example3.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 3 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 6]
```diff
@@ -175,8 +176,9 @@ public int[] getRequiredTokens() {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final DetailAST leftCurly = findLeftCurly(ast);
-        if (leftCurly != null) {
+        final Optional<DetailAST> lcurlyOptional = findLeftCurly(ast);
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:179]: Please never name Optional variables as `Optional...`. We force the developer to think about the semantics of this type by contract.

```suggestion
        final Optional<DetailAST> leftCurly = findLeftCurly(ast);
```

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 7]
```diff
@@ -270,23 +272,28 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
...
+        if (parentType == TokenTypes.SWITCH_RULE) {
+            lcurlyOptional = Optional.ofNullable(parent.findFirstToken(TokenTypes.SLIST));
         }
-        else if (slistAST == null) {
-            leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+        else if (parentType == TokenTypes.CASE_GROUP) {
+            final DetailAST statementList = ast.getNextSibling();
+            lcurlyOptional = Optional.ofNullable(statementList)
+                         .map(DetailAST::getFirstChild)
+                         .filter(node -> node.getType() == TokenTypes.SLIST);
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:287]: This can be one assignment, I am not sure what we gain by having this extra variable declaration

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 8]
```diff
@@ -270,23 +272,28 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
...
         }
         else {
-            leftCurly = slistAST;
+            DetailAST leftCurly = ast.findFirstToken(TokenTypes.SLIST);
+            if (leftCurly == null) {
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:291]: This can be separated into another `else if`, please do this (via Optional#of since we will now have a not null check); also make a comment in each branch here about what type we expect in each.

Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:291]:
```
        else if (ast.findFirstToken(TokenTypes.SLIST) != null) {
            lcurlyOptional = Optional.of(ast.findFirstToken(TokenTypes.SLIST));
        }
```

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 9]
```diff
@@ -270,23 +272,28 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
...
+                leftCurly = ast.findFirstToken(TokenTypes.LCURLY);
+            }
+            lcurlyOptional = Optional.ofNullable(leftCurly);
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:294]: If we do ^, then you end up with a very succinct catch all:
```suggestion
        else {
            lcurlyOptional = Optional.ofNullable(ast.findFirstToken(TokenTypes.LCURLY));
        }
```

This makes a lot of sense; if we match none of the special cases above, we just try to get the first left curly that exists if it is present.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java [hunk 10]
```diff
@@ -270,23 +272,28 @@ private boolean checkIsAllLinesAreWhitespace(int lineFrom, int lineTo) {
...
-    private static DetailAST findLeftCurly(DetailAST ast) {
-        final DetailAST leftCurly;
-        final DetailAST slistAST = ast.findFirstToken(TokenTypes.SLIST);
-        if ((ast.getType() == TokenTypes.LITERAL_CASE
-                || ast.getType() == TokenTypes.LITERAL_DEFAULT)
-                && ast.getNextSibling() != null
-                && ast.getNextSibling().getFirstChild() != null
-                && ast.getNextSibling().getFirstChild().getType() == TokenTypes.SLIST) {
-            leftCurly = ast.getNextSibling().getFirstChild();
+    private static Optional<DetailAST> findLeftCurly(DetailAST ast) {
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:275]:
```suggestion
    private static Optional<DetailAST> getLeftCurly(DetailAST ast) {
```

Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java:275]: `find` doesn't imply `get` :)

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 10 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: good catch :). done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 9 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: > This makes a lot of sense; if we match none of the special cases above, we just try to get the first left curly that exists if it is present.

I agree, done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 8 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: good, done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 7 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: yea done didn't add much value

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 6 above)
Comment (mahfouz72) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: done. renamed optional to leftCurly and the value we get from the Optional to leftCurlyAST is it good ?

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Last from me: please add a test case for `text` block option, where we have some comments explaining why we have no actual code.

After you push your next changes, let's generate the report again

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java (hunk 6 above)
Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java]: yep

**Comment by mahfouz72** [comment]
github, generate report

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java (hunk 1 above)
Comment (nrmancuso) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java:116]: > Text was added in #8662 . `CASE` was a token of the check then, so I am slightly confused why we originally said this was ok to be allowed.
> 
> @nrmancuso When you review.

Yeah this was weird, I think this should be a violation

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java [hunk 11]
```diff
@@ -0,0 +1,94 @@
+/*
+EmptyBlock
+option = text
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java:3]: Example using statement.

**Comment by rnveach** [comment]
@nrmancuso 
> What is a sequential block?
I have no idea. The doc for this check is not good should we update it? in another PR?

Do you have any ideas about this? It is specifically written in the documentation for this check.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+tokens = LITERAL_DEFAULT, LITERAL_CASE, LITERAL_SWITCH
```
Comment (rnveach) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java:4]: More curious. Why did you add `switch` here, when you don't add it in the updated example.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java (hunk 11 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java:3]: We have this already (The input file that I modifed in this PR)
https://github.com/checkstyle/checkstyle/blob/3369a7ab5dd8e07d5299e6016a88e38e0cba09a9/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockSwitchExpressions.java

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java (hunk 12 above)
Comment (mahfouz72) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockCaseAndDefaultWithTextOption.java]: Copy paste from another file and forget to remove it. but actually, we don't need it here, removed

**Comment by nrmancuso** [comment]
> @nrmancuso
> 
> > What is a sequential block?
> > I have no idea. The doc for this check is not good should we update it? in another PR?
> 
> Do you have any ideas about this? It is specifically written in the documentation for this check.

I have no idea what this means, I am sorry 😂

**Comment by rnveach** [comment]
@mahfouz72 @nrmancuso 
Text was added at https://github.com/checkstyle/checkstyle/commit/350801597ccce93d019ab03f5f171d5acc27e3d9 . Looks like conversation started at https://github.com/checkstyle/checkstyle/pull/3918#discussion_r107571758 .

It is meant to say we don't support `case 0: {} {}`.

**Review by rnveach — APPROVED** [review]
